src/pydvl/valuation/methods/classwise_shapley.py

```diff
@@ -89,75 +83,20 @@
 
 logger = logging.getLogger(__name__)
 
-
-def unique_labels(array: NDArray) -> NDArray:
-    """Labels of the dataset."""
-    # Object, String, Unicode, Unsigned integer, Signed integer, boolean
-    if array.dtype.kind in "OSUiub":
-        return np.unique(array)
-    raise ValueError("Dataset must be categorical to have unique labels.")
-
-
-class ClasswiseSampler(IndexSampler):
-    def __init__(
-        self,
-        in_class: IndexSampler,
-        out_of_class: PowersetSampler,
-        label: int | None = None,
-    ):
-        super().__init__()
-        self.in_class = in_class
-        self.out_of_class = out_of_class
-        self.label = label
-
-    def for_label(self, label: int) -> ClasswiseSampler:
-        return ClasswiseSampler(self.in_class, self.out_of_class, label)
-
-    def from_data(self, data: Dataset) -> Generator[list[CSSample], None, None]:
-        assert self.label is not None
-
-        without_label = np.where(data.y != self.label)[0]
-        with_label = np.where(data.y == self.label)[0]
-
-        # HACK: the outer sampler is over full subsets of T_{-y_i}
-        self.out_of_class._index_iteration = NoIndexIteration
-
-        for ooc_batch in self.out_of_class.generate_batches(without_label):
-            # NOTE: The inner sampler can be a permutation sampler => we need to
-            #  return batches of the same size as that sampler in order for the
-            #  in_class strategy to work correctly.
-            for ooc_sample in ooc_batch:
-                for ic_batch in self.in_class.generate_batches(with_label):
-                    # FIXME? this sends the same out_of_class_subset for all samples
-                    #   maybe a few 10s of KB... probably irrelevant
-                    yield [
-                        CSSample(
-                            idx=ic_sample.idx,
-                            label=self.label,
-                            subset=ooc_sample.subset,
-                            in_class_subset=ic_sample.subset,
-                        )
-                        for ic_sample in ic_batch
-                    ]
-
-    def generate_batches(self, indices: IndexSetT) -> BatchGenerator:
-        raise AttributeError("Cannot sample from indices directly.")
-
-    def make_strategy(
-        self,
-        utility: UtilityBase,
-        coefficient: Callable[[int, int], float] | None = None,
-    ) -> EvaluationStrategy[IndexSampler]:
-        return self.in_class.make_strategy(utility, coefficient)
+T = TypeVar("T")
 
 
 class ClasswiseShapley(Valuation):
+    algorithm_name = "Classwise-Shapley"
+
```

Comment:
This class should get a docstring referencing the details of the method in the docs.